crates/bevy_ui/src/widget/button.rs

```diff
@@ -1,7 +1,13 @@
+use crate::{FocusPolicy, Interaction, Node};
 use bevy_ecs::{prelude::Component, reflect::ReflectComponent};
 use bevy_reflect::{std_traits::ReflectDefault, Reflect};
 
 /// Marker struct for buttons
 #[derive(Component, Debug, Default, Clone, Copy, PartialEq, Eq, Reflect)]
 #[reflect(Component, Default, Debug, PartialEq)]
+#[require(Node, FocusPolicy(block_focus_policy), Interaction)]
 pub struct Button;
+
+fn block_focus_policy() -> FocusPolicy {
```

Comment:
This should be defined inline as

```rust
#[require(FocusPolicy(|| FocusPolicy::Block))]
```

Reply:
Thank! I didn't realize I could use closure. Fixed.

Should use [In-line Required Component Values](https://hackmd.io/@bevy/required_components/%2FHwaPiH3GRWqxrEkCKRzXYQ) when it's completed.